ethcore/src/client/client.rs

```diff
@@ -771,56 +624,290 @@ impl Client {
 								Ok(res) => Ok((res.output, state.drop().1.extract_proof())),
 							};
 
-							res.map(|(output, proof)| (output, proof.into_iter().map(|x| x.into_vec()).collect()))
-						};
+							res.map(|(output, proof)| (output, proof.into_iter().map(|x| x.into_vec()).collect()))
+						};
+
+						match with_state.generate_proof(&call) {
+							Ok(proof) => proof,
+							Err(e) => {
+								warn!(target: "client", "Failed to generate transition proof for block {}: {}", hash, e);
+								warn!(target: "client", "Snapshots produced by this client may be incomplete");
+								Vec::new()
```

Comment:
if `client` failed to generate `proof`, why do we insert an invalid proof (`Vec::new()`) into chain? line 644

Reply:
nvm. I see that this code hasn't been changed in this pr

Reply:
to answer the original question, I was a bit ambivalent about what to do. It indicates that either the validator contract or the local database are broken so whatever we do the chain is going to break at some point.